sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryStorageStreamSource.java

```diff
@@ -322,5 +332,23 @@ public synchronized void close() {
           "Successfully split BigQuery Storage API stream. Split response: {}", splitResponse);
       return source.fromExisting(splitResponse.getRemainderStream());
     }
+
+    @Override
+    public synchronized Double getFractionConsumed() {
+      return fractionConsumed;
+    }
+
+    private static float getFractionConsumed(ReadRowsResponse response) {
+      // TODO(aryann): Once we rebuild the generated client code, we should change this to
+      // use getFractionConsumed().
```

Comment:
We should be sure to update the client code before 2.15 is released. The use of UnknownField types is probably not appropriate for production code.

Reply:
My thinking is that since our primary use case is not to be a reference implementation that this is okay, though I do think that sometime soon we should update the client.

I'm not sure if it makes sense to block 2.15 on the client being updated, though. The comments I've added make it clear what the fields we are setting are for someone who is trying to understand the internals of our source.

That said, I'm willing to change my mind if there is a stronger opinion on this. :)